extension/background/serviceList.js

```diff
@@ -164,14 +164,29 @@ export class Service {
 }
 
 export async function detectServiceFromActiveTab(services) {
+  let serviceName = null;
   const tab = await browserUtil.activeTab();
   for (const name in services) {
     const service = services[name];
     if (tab.url.startsWith(service.baseUrl)) {
-      return name;
+      if (service.necessaryQuery === undefined) {
+        serviceName = name;
+        continue;
```

Comment:
And then I might leave a comment here: even if this matches, we still want to check if there's another more specific search provider that matches with query strings